plugins/woo-ai/src/utils/productData.ts

```diff
@@ -102,8 +102,16 @@ const getProductType = () => {
 	return productTypeEl ? productTypeEl.value : '';
 };
 
+const getProductId = (): number => {
+	const productIdEl: HTMLInputElement | null =
+		document.querySelector( '#post_ID' );
+
+	return productIdEl ? parseInt( productIdEl.value, 10 ) : 0;
```

Comment:
It looks like since this defaults to `0`, which is a number, this would result in the product slug hook still trying to update a post with an ID of zero? 

You might return `null` here, and then just abandon changing the slug in that case.